File: scale-x-y/scale_xy.lua
io.write("Enter x coordinate: ")
x= io.read()

io.write("Enter y coordinate: ")
y= io.read()

io.write("Enter x coordinate of origin: ")
origin_x= io.read()

io.write("Enter y coordinate of origin: ")
origin_y= io.read()

io.write("Enter scale factor: ")
scale = io.read()

new_x = scale * (x - origin_x) + origin_x
new_y = scale * (y - origin_y) + origin_y

print(new_x .. ", " .. new_y)

File: scale-x-y/scale_xy.py
x= int(input("Enter x coordinate: "))
y= int(input("Enter y coordinate: "))

origin_x= int(input("Enter x coordinate of origin: "))
origin_y= int(input("Enter y coordinate of origin: "))

scale = int(input("Enter scale factor: "))

new_x = scale * (x - origin_x) + origin_x
new_y = scale * (y - origin_y) + origin_y

print(f"{new_x}, {new_y}")